### PR TITLE
Removing rhf input dependencie

### PR DESCRIPTION
**Pull request by ardguezsoc**
3c8ee82:

-  rhf 6.14.2 --> 7.1.2  
- adding react hooks for new logic testing necessary to create useForm() object to make a control object from it.

93fbe2b:
- erros now come from formState 
- new format to register components ref -- > {...register}

206f388:
- rhfi dependencie removed and now we will use the controller component from rhf that is a wrapper. We will need the control object too for this kind of components.

e69a3db:

- Due the new logic error needed to be changed and control object needed to be mocked for testing

### CODE REVIEW

File: src/builder.js [hunk 1]
```diff
@@ -190,7 +190,7 @@ const FormBuilder = ({
   const formatData = (data) => {
     Object.keys(data).map((key) => {
       if (data[key] instanceof Date) {
-        data[key] = data[key].toISOString()
+        data[key] = new Date(data[key] + 'Z').toISOString()
```
Comment (ismaelocaramelo) [inline src/builder.js:193]: Have we test this?

**Review by ismaelocaramelo — COMMENTED** [review]
Seems a big change, but not sure if only with integration test are we 100% safe.

File: src/builder.js (hunk 1 above)
Comment (ardguezsoc) [reply src/builder.js:193]: yes

**Comment by ismaelocaramelo** [comment]
Same, this has some conflicts @ardguezsoc

**Comment by inigomarquinez** [comment]
Closing this PR as it has been in a stale status for a long time and the library has lot of changes since then and a lot of conflicts have appeared.

**Pull request closed without merging**